ctk/common/src/test/kotlin/org/codice/compilance/verification/core/CommonDataTypeVerifierSpec.kt

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compilance.verification.core
+
+import io.kotlintest.forAll
+import io.kotlintest.matchers.string.shouldContain
+import io.kotlintest.shouldThrow
+import io.kotlintest.specs.StringSpec
+import org.codice.compliance.Common.Companion.buildDom
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLCore_1_3_1_a
+import org.codice.compliance.SAMLCore_1_3_2_a
+import org.codice.compliance.SAMLCore_1_3_3_a
+import org.codice.compliance.SAMLCore_1_3_4_a
+import org.codice.compliance.SAMLCore_3_4_1_1_a
+import org.codice.compliance.SAMLSpecRefMessage
+import org.codice.compliance.verification.core.CommonDataTypeVerifier
+import org.w3c.dom.Node
+
+@Suppress("StringLiteralDuplication" /* Small duplicate test strings */)
+class CommonDataTypeVerifierSpec : StringSpec() {
+    init {
+        "null or blank string value fails" {
+            badString(null)
+            badString(null, SAMLCore_3_4_1_1_a)
+            badString("")
+            badString("   ")
+            badString("   ", SAMLCore_3_4_1_1_a)
+        }
+
+        "valid string value passes" {
+            forAll(listOf("a", "abc", "whatever\nyou say")) {
+                CommonDataTypeVerifier.verifyStringValues(buildDom("<fld>$it</fld>"))
+                CommonDataTypeVerifier.verifyStringValues(buildDom("<fld>$it</fld>"),
+                        SAMLCore_3_4_1_1_a)
+            }
+        }
+
+        "bad Uri value fails" {
+            badUri(null)
+            badUri(null, SAMLCore_3_4_1_1_a)
+            badUri("")
+            badUri("   ")
+            badUri("/not/absolute")
+            badUri("/not/absolute", SAMLCore_3_4_1_1_a)
```

Comment:
✏️ Could you use the `forAll` to test these combinations like you did above?

Reply:
Good idea. 👍